### PR TITLE
Telegraf

### PR DESCRIPTION
**Pull request by laullon**

### CODE REVIEW

File: plugins/sources/telegraf/provider.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+	var sources []wf.MetricsSource
+	for name, creator := range telegrafInputs.Inputs {
+		plugin := creator()
+		if name == "procstat" {
+			continue // I hate me ;-)
+		}
+		sources = append(sources, newTelegrafPluginSource(name+" plugin", plugin))
+	}
+
+	for _, exe := range []string{"wavefront-collector", "dockerd"} {
```
Comment (vikramraman) [inline plugins/sources/telegraf/provider.go]: can remove

File: plugins/sources/telegraf/provider.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+	for _, pair := range os.Environ() {
+		glog.Infof("env: %v", pair)
+	}
+
+	glog.Infof("[telegraf.NewProvider] - inputs: %v -------------", telegrafInputs.Inputs)
+
+	var sources []wf.MetricsSource
+	for name, creator := range telegrafInputs.Inputs {
+		plugin := creator()
+		if name == "procstat" {
```
Comment (vikramraman) [inline plugins/sources/telegraf/provider.go]: let's leave this out for now

File: plugins/sources/telegraf/provider.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+func NewProvider(uri *url.URL) (wf.MetricsSourceProvider, error) {
+
+	for _, pair := range os.Environ() {
+		glog.Infof("env: %v", pair)
+	}
+
+	glog.Infof("[telegraf.NewProvider] - inputs: %v -------------", telegrafInputs.Inputs)
+
+	var sources []wf.MetricsSource
+	for name, creator := range telegrafInputs.Inputs {
```
Comment (vikramraman) [inline plugins/sources/telegraf/provider.go]: Can we make this configurable? If `telegraf:''` we can default to including all inputs. But also provide user an option to specify what they want. We'd need some kind of mapping/sanitizing for the input name.

File: plugins/sources/telegraf/provider.go [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+	mux    sync.Mutex
+}
+
+func newTelegrafPluginSource(name string, plugin telegraf.Input) *telegrafPluginSource {
+	hostname := os.Getenv("POD_NODE_NAME")
+	glog.Infof("hostname: '%s'", hostname)
+
+	tsp := &telegrafPluginSource{name: name, plugin: plugin, source: hostname}
+	ticker := time.NewTicker(5 * time.Second)
+	go func() {
```
Comment (vikramraman) [inline plugins/sources/telegraf/provider.go]: remove? Is this just for debugging?

File: plugins/sources/telegraf/provider.go [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+		for {
+			select {
+			case <-ticker.C:
+				err := plugin.Gather(tsp)
```
Comment (vikramraman) [inline plugins/sources/telegraf/provider.go]: We'd want to do the actual gathering within the `ScrapeMetrics` function at the intervals configured on the collector level.

File: plugins/sources/telegraf/provider.go [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+func (p telegrafProvider) Name() string {
+	return "Telegraf Source"
+}
+
+type telegrafPluginSource struct {
+	name   string
+	source string
+	plugin telegraf.Input
+	points []*wf.MetricPoint
```
Comment (vikramraman) [inline plugins/sources/telegraf/provider.go]: We shouldn't need a local reference for points. Collect / return them from within the `ScrapeMetrics`?

File: plugins/sources/telegraf/provider.go (hunk 4 above)
Comment (laullon) [reply plugins/sources/telegraf/provider.go]: The log msg ?

**Comment by vikramraman** [comment]
Thanks for doing the grunt work on this! Closing this PR in favor of #25  which includes this and other changes.

**Pull request closed without merging**